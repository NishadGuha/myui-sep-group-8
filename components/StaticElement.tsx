import React from "react";

/**
 * Convert text to media 
 *
 * @param {string} text
 * @return {*}  {JSX.Element}
 */
function convertText(text: string): JSX.Element {
  // Check if the text ends with a media extension
  if (text.endsWith('.png') || text.endsWith('.jpg') || text.endsWith('.jpeg') || text.endsWith('.gif')) {
    return (
      // Display the image
      <img height={"100%"} width={"100%"} src={text} />
    );
  } else if (text.endsWith('.mp4') || text.endsWith('.mp3')) {
    return (
      // Display the video
      <video height={"100%"} width={"100%"} src={text} controls/>
    );
  } else {
    return (
      // Display the text
      <p style={{height:"100%", width:"100%", overflow:"auto"}}>{text}</p>
    );
  }
}

/**
 * @param {*} {text, style}
 * @return {*}  {JSX.Element}
 */
export default function StaticElement({text, style}: any): JSX.Element {
  return (
    <div style={style}>
      {convertText(text)}
    </div>
  );
};
