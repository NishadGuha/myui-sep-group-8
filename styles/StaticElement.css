.dashboard-text {
  white-space: pre-line;
  font-size: large;
  height: 100%;
  width: 100%;
  overflow: auto;
}
