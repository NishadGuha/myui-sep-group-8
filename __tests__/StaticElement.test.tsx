import { render, screen } from "@testing-library/react";
import React from "react";
import renderer from "react-test-renderer";
import StaticElement from "../components/StaticElement";
import "@testing-library/jest-dom";

describe("Static element adapts to content", () => {
  it("Renders text correctly", () => {
    render(<StaticElement 
            text={"Some test text!"} 
            mediaDisplaySetting={"MEDIA"}
          />);
    const renderedElement = screen.getByTestId("static-element");
    expect(renderedElement.firstChild).toBeInstanceOf(HTMLParagraphElement);
  })

  it("Renders png image correctly", () => {
    render(<StaticElement 
            text={"Some_img_url.png"} 
            mediaDisplaySetting={"MEDIA"}
          />);
    const renderedElement = screen.getByTestId("static-element");
    expect(renderedElement.firstChild).toBeInstanceOf(HTMLImageElement);
  })

  it("Renders jpg image correctly", () => {
    render(<StaticElement 
            text={"Some_img_url.jpg"} 
            mediaDisplaySetting={"MEDIA"}
          />);
    const renderedElement = screen.getByTestId("static-element");
    expect(renderedElement.firstChild).toBeInstanceOf(HTMLImageElement);
  })

  it("Renders jpeg image correctly", () => {
    render(<StaticElement 
            text={"Some_img_url.jpeg"} 
            mediaDisplaySetting={"MEDIA"}
          />);
    const renderedElement = screen.getByTestId("static-element");
    expect(renderedElement.firstChild).toBeInstanceOf(HTMLImageElement);
  })

  it("Renders gif correctly", () => {
    render(<StaticElement 
            text={"Some_img_url.gif"} 
            mediaDisplaySetting={"MEDIA"}
          />);
    const renderedElement = screen.getByTestId("static-element");
    expect(renderedElement.firstChild).toBeInstanceOf(HTMLImageElement);
  })

  it("Renders mp3 correctly", () => {
    render(<StaticElement 
            text={"Some_img_url.mp3"} 
            mediaDisplaySetting={"MEDIA"}
          />);
    const renderedElement = screen.getByTestId("static-element");
    expect(renderedElement.firstChild).toBeInstanceOf(HTMLVideoElement);
  })

  it("Renders mp4 correctly", () => {
    render(<StaticElement 
            text={"Some_img_url.mp4"} 
            mediaDisplaySetting={"MEDIA"}
          />);
    const renderedElement = screen.getByTestId("static-element");
    expect(renderedElement.firstChild).toBeInstanceOf(HTMLVideoElement);
  })
})

describe("Snapshot tests for static element adapts to content", () => {
  it("Renders text correctly", () => {
    const tree = renderer
      .create(<StaticElement 
        text={"Some test text!"} 
        mediaDisplaySetting={"MEDIA"}
      />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  })

  it("Renders png image correctly", () => {
    const tree = renderer
      .create(<StaticElement 
              text={"Some_img_url.png"} 
              mediaDisplaySetting={"MEDIA"}
            />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  })

  it("Renders jpg image correctly", () => {
    const tree = renderer
      .create(<StaticElement 
              text={"Some_img_url.jpg"} 
              mediaDisplaySetting={"MEDIA"}
            />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  })

  it("Renders jpeg image correctly", () => {
    const tree = renderer
      .create(<StaticElement 
                text={"Some_img_url.jpeg"} 
                mediaDisplaySetting={"MEDIA"}
              />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  })

  it("Renders gif correctly", () => {
    const tree = renderer
      .create(<StaticElement 
                text={"Some_img_url.gif"} 
                mediaDisplaySetting={"MEDIA"}
              />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  })

  it("Renders mp3 correctly", () => {
    const tree = renderer
      .create(<StaticElement 
                text={"Some_img_url.mp3"} 
                mediaDisplaySetting={"MEDIA"}
              />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  })

  it("Renders mp4 correctly", () => {
    const tree = renderer
      .create(<StaticElement 
                text={"Some_img_url.mp4"} 
                mediaDisplaySetting={"MEDIA"}
              />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  })
})