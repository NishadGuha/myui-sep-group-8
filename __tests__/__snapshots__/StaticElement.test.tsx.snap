// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot tests for static element adapts to content Renders gif correctly 1`] = `
<div
  data-testid="static-element"
>
  <img
    height="100%"
    src="Some_img_url.gif"
    width="100%"
  />
</div>
`;

exports[`Snapshot tests for static element adapts to content Renders jpeg image correctly 1`] = `
<div
  data-testid="static-element"
>
  <img
    height="100%"
    src="Some_img_url.jpeg"
    width="100%"
  />
</div>
`;

exports[`Snapshot tests for static element adapts to content Renders jpg image correctly 1`] = `
<div
  data-testid="static-element"
>
  <img
    height="100%"
    src="Some_img_url.jpg"
    width="100%"
  />
</div>
`;

exports[`Snapshot tests for static element adapts to content Renders mp3 correctly 1`] = `
<div
  data-testid="static-element"
>
  <video
    controls={true}
    controlsList="nofullscreen"
    height="100%"
    src="Some_img_url.mp3"
    width="100%"
  />
</div>
`;

exports[`Snapshot tests for static element adapts to content Renders mp4 correctly 1`] = `
<div
  data-testid="static-element"
>
  <video
    controls={true}
    controlsList="nofullscreen"
    height="100%"
    src="Some_img_url.mp4"
    width="100%"
  />
</div>
`;

exports[`Snapshot tests for static element adapts to content Renders png image correctly 1`] = `
<div
  data-testid="static-element"
>
  <img
    height="100%"
    src="Some_img_url.png"
    width="100%"
  />
</div>
`;

exports[`Snapshot tests for static element adapts to content Renders text correctly 1`] = `
<div
  data-testid="static-element"
>
  <p
    className="dashboard-text"
  >
    Some test text!
  </p>
</div>
`;
